modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/OAuthConnectionProviderObjectBuilder.java

```diff
@@ -237,7 +256,10 @@ private void withCustomParameters(BiConsumer<ParameterModel, OAuthParameterModel
   }
 
   private String resolveString(CoreEvent event, ValueResolver resolver) throws MuleException {
-    Object value = resolver.resolve(from(event));
+    Object value = resolver.resolve(ValueResolvingContext.builder(event)
+        .dynamic(resolver.isDynamic())
```

Comment:
Is quite weird that the resolver needs a `ValueResolvingContext` with information of himself instead of doing automatically.
This gives the developer chances of creating no sense scenarios.